AdminPortal/src/pages/dashboard.tsx

```diff
@@ -1,6 +1,284 @@
-/**
- * Renders the {url}/dashboard page.
- */
-export default function Dashboard() {
-  return <div>{'This is the dashboard!'}</div>;
+import React, { useState, useEffect } from 'react';
+import styles from '@/styles/Dashboard.module.css';
+import Dropdown from 'react-bootstrap/Dropdown';
+import 'bootstrap/dist/css/bootstrap.min.css';
+import data from '../data/data.json';
+import { Lato } from '@next/font/google';
+import { Sidebar } from '@/components/sidebar';
+import Select, { InputActionMeta } from 'react-select';
+
+interface DataItem {
+  id: number;
+  dateApplied: string;
+  name: string;
+  mobileNumber: string;
+  licenseID: string;
+  licensePlate: string;
+  isChecked: boolean;
+  category: String;
+}
+
+interface Option {
+  label: string;
+  value: string;
+}
+
+interface ControlStyles {
+  [key: string]: unknown;
+}
+
+export default function App() {
+  const numTabs = 4;
+
+  const [activeTab, setActiveTab] = useState<string>('Needs Review');
+
+  const [selectAllClicked, setSelectAllClicked] = useState(
+    new Array(numTabs).fill(false)
+  );
+
+  // for the dropdown
+  const [selected, setSelected] = useState<Option | null>(null);
+
+  // mapping the tabs to an integer key
+  const tabMapping = new Map();
+  tabMapping.set('Needs Review', 1);
+  tabMapping.set('In Review', 2);
+  tabMapping.set('Verified', 3);
+  tabMapping.set('Disapproved', 4);
```

Comment:
What's the use of this mapping here?

Reply:
it is essentially used for indexing at places, converting tabs to a numeric value